src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -26,6 +26,12 @@ public class ProgressTracker
         private int _activeCodeRequests;
         private int _activeAccRefreshRequests;
 
+        // Speed calculation is a bit broken, so we don't set this too high until we fix that to prevent getting stuck
+        // on slow peer which might reduces overall overall throughput. This is still an improvement though.
+        public const int MAX_ACCOUNT_REQUEST_WAIT = 200;
+        private readonly TimeSpan _maxAccountRequestWait = TimeSpan.FromMilliseconds(MAX_ACCOUNT_REQUEST_WAIT);
+        private readonly AutoResetEvent _accountRequestCompleted = new(false);
```

Comment:
Signaling with AutoResetEvent is much slower than for example with Monitor.Pulse but the latter is less readable if one wants to do it right. 
So maybe we should keep AutoResetEvent.
We can think about it later ... :)